protostar/configuration_file/configuration_file_v2.py

```diff
@@ -89,5 +96,48 @@ def read(
     ) -> ConfigurationFileV2Model:
         raise NotImplementedError("Operation not supported.")
 
-    def save(self, configuration_file_model: ConfigurationFileV2Model) -> Path:
-        raise NotImplementedError("Operation not supported.")
+    def create_file_content(
```

Comment:
I wonder if it wouldn't be simpler to simply build a Python dict containing all data and pass it to the writer.

In general that would drop the need for the `ConfigurationFileContentBuilder` class.

Reply:
Shouldn't `profile` en/decoding be shifted to a separate class that would be layered over all of this? You're making too powerful objects here.

Reply:
- there might be different conventions for encoding `profiles` per format 
- `ConfigurationTOMLContentBuilder` has 54 lines of cohesive code
- there are more important things to do